### ISSUE TITLE
Exponential backoff for -queryserver-config-txpool-timeout

### ISSUE THREAD

**Issue by schneiderdba** [vitessio/vitess#5035]
There is a vttablet configurable "queryserver-config-txpool-timeout" that controls the query server transaction pool timeout and how long vttablet waits if tx pool is full.  

#### Feature Description

It would be nice to have the ability to define an exponential backoff instead of a static timed wait.  

#### Use Case(s)

Being able to define an exponential backoff may be useful when transaction pool resources become overwhelmed.

**Comment by dweitzman** [comment]
The txpool timeout represents the maximum wait time. If a transaction frees up before that time expires, its connection will become immediately available for use and unblock one of the waiters.

This seems mostly better than backing off by sleeping since there won't be any dead time when connections are available but not being used (unless for some reason the Go scheduler fails to wake up a goroutine and let it receive a transaction from the free transactions channel).

**Comment by sougou** [comment]
The backoff needs to be done at the app layer. The vttablet layer should actually do something opposite: to fail early if there's no hope of getting a connection in time. Facebook published this paper on it: https://queue.acm.org/detail.cfm?id=2839461. It's something we can look at adopting.
